app/retail/templates/emails/cart.html

```diff
@@ -0,0 +1,79 @@
+{% extends 'emails/template.html' %}
+{% comment %}
+  Copyright (C) 2020 Gitcoin Core
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU Affero General Public License as published
+  by the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU Affero General Public License for more details.
+
+  You should have received a copy of the GNU Affero General Public License
+  along with this program. If not, see <http://www.gnu.org/licenses/>.
+{% endcomment %}
+{% load i18n humanize %}
+
+{% block content %}
+
+<style>
+  .arrow-up{
+    width: 0;
+    height: 0;
+    border-left: 15px solid transparent;
+    border-right: 15px solid transparent;
+    border-bottom: 15px solid #ddd;
+    margin-left: auto;
+    margin-right: auto;
+  }
+
+  #founder_announce {
+    background-color: #fafafa;
+    margin-bottom: 15px;
+  }
+
+  #founder_announce a{
+    font-weight: bold;
+  }
+</style>
+
+<div>
+  <h1 class="text-centre">{% trans "Your cart is waiting for you" %}</h1>
+</div>
+
+<a href="https://gitcoin.co/owocki">
+<img style="width: 100px; height: 100px;" src="https://gitcoin.co/dynamic/avatar/owocki">
+</a>
+<div class="arrow-up">&nbsp;</div>
+<div id="founder_announce" style="border: 1px solid #ddd; max-width: 500px; text-align: center; margin-left: auto; margin-right: auto; margin-bottom: 25px;">
+<p style="margin: 5px; border-radius: 5px;">
+  {{desc|safe}}
+</p>
+</div>
+
+<HR>
+
+<div>
+  <p style="margin: 5px; border-radius: 5px;">
+    You have in your cart the following grants:
+  </p>
+  <ol>
+    {% for grant in grants %}
+      <li><b>{{ grant.grant_title }}</b> ({{ grant.grant_donation_amount|default:5 }} {{ grant.grant_donation_currency|default:"DAI" }})</li>
+    {% endfor %}
+  </ol>
+</div>
+
+<HR>
+
+<a href={{base_url}}>
+  <div style="margin-bottom: 2em; margin-top: 2em;">
+    <a class="button large" href="{{base_url}}{%  url 'grants:grants_bulk_add' cart_query %}">{% trans "Go to cart" %}</a>
```

Comment:
```
Traceback (most recent call last):
  File "./manage.py", line 59, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/base.py", line 323, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/base.py", line 364, in execute
    output = self.handle(*args, **options)
  File "/code/app/grants/management/commands/sent_cart_reminder.py", line 91, in handle
    remember_your_cart(activity.profile, cart_query, no_checkout_grants, hours)
  File "/code/app/marketing/mails.py", line 1915, in remember_your_cart
    html, text = render_remember_your_cart(cart_query, grants, hours)
  File "/code/app/retail/emails.py", line 1618, in render_remember_your_cart
    response_html = premailer_transform(render_to_string("emails/cart.html", params))
  File "/usr/local/lib/python3.6/dist-packages/django/template/loader.py", line 62, in render_to_string
    return template.render(context, request)
  File "/usr/local/lib/python3.6/dist-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
  File "/usr/local/lib/python3.6/dist-packages/django/template/base.py", line 171, in render
    return self._render(context)
  File "/usr/local/lib/python3.6/dist-packages/django/test/utils.py", line 96, in instrumented_test_render
    return self.nodelist.render(context)
  File "/usr/local/lib/python3.6/dist-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.6/dist-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.6/dist-packages/django/template/loader_tags.py", line 150, in render
    return compiled_parent._render(context)
  File "/usr/local/lib/python3.6/dist-packages/django/test/utils.py", line 96, in instrumented_test_render
    return self.nodelist.render(context)
  File "/usr/local/lib/python3.6/dist-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.6/dist-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.6/dist-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/usr/local/lib/python3.6/dist-packages/django/template/base.py", line 937, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.6/dist-packages/django/template/base.py", line 904, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.6/dist-packages/django/template/defaulttags.py", line 443, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
  File "/usr/local/lib/python3.6/dist-packages/wiki/models/__init__.py", line 35, in reverse
    url = original_django_reverse(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/django/urls/base.py", line 90, in reverse
    return iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs))
  File "/usr/local/lib/python3.6/dist-packages/django/urls/resolvers.py", line 668, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'grants_bulk_add' with arguments '('',)' not found. 1 pattern(s) tried: ['grants\\/cart\\/bulk\\-add\\/(?P<grant_str>[^/]+)$']
```

Reply:
maybe add a try / catch?

Reply:
or just dont use the reverse path, or check for false'ey carts?

Reply:
Added validation and one try/catch